zjit/src/codegen.rs

```diff
@@ -762,12 +762,23 @@ fn gen_ccall_variadic(
     let stack_growth = state.stack_size();
     gen_stack_overflow_check(jit, asm, state, stack_growth);
 
+    let block_handler_specval = if let Some(block_iseq) = blockiseq {
```

Comment:
I don't like repeating this again but I want to take more time to understand specval before doing any refactor around it, so I opened https://github.com/Shopify/ruby/issues/827